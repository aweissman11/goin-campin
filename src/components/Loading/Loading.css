.loading {
  box-sizing: border-box;
  object-fit: contain;
  height: stretch;
  background: #3f6358;
}

.gif-container {
  height: 90vh;
  position: relative;
}

.loading-message {
  background: #3f6358;
  z-index: 1403;
}

.loading-gif {
  width: 100%;
  height: stretch;
  background: #3f6358;

}

.gradient {
  background: linear-gradient(0deg, rgba(0, 0, 0, 0), #3f6358);
  position: absolute;
  top: 0;
  left: 0;
  width: 100vw;
  height: stretch;
  z-index: 120;
}