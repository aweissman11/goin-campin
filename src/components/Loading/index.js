import React from 'react'

import './Loading.css'

const Loading = (props) => {
  return(
    <div className='loading'>
      <h2 className='loading-message'>{props.message}</h2>
      <div className='gif-container'>
        <img 
          alt='loading'
          className='loading-gif'
          src={`https://cdn.dribbble.com/users/2524821/screenshots/5358445/untitled-1.gif`}
        />
        <div className='gradient'></div>
      </div>
    </div>
  );
};

export default Loading;