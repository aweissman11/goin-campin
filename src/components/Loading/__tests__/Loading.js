import React from 'react';
import { shallow } from 'enzyme';

import Loading from '../index';

describe('Loading', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Loading message='loading message'/>)
    expect(wrapper).toMatchSnapshot()
  })
  
})
