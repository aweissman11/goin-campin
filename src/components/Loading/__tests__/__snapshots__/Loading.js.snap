// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loading
    message="loading message"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          loading message
        </h2>,
        <img
          src="https://cdn.dribbble.com/users/2524821/screenshots/5358445/untitled-1.gif"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "loading message",
        },
        "ref": null,
        "rendered": "loading message",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "https://cdn.dribbble.com/users/2524821/screenshots/5358445/untitled-1.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            loading message
          </h2>,
          <img
            src="https://cdn.dribbble.com/users/2524821/screenshots/5358445/untitled-1.gif"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "loading message",
          },
          "ref": null,
          "rendered": "loading message",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "https://cdn.dribbble.com/users/2524821/screenshots/5358445/untitled-1.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
